src/Uno.UI/DirectUI/TrackableDateCollection.cs

```diff
@@ -137,6 +137,12 @@ public override void InsertAt(uint index, DateTime item)
 			return;
 		}
 
+		public override void Add(DateTime item)
+		{
+			DateTime newItem = new DateTime(item.Date).ToUniversalTime();
+			Append(newItem.Add(item.Date - newItem.Date));
+		}
```

Comment:
I think this logic should be on the problematic callsite. If I have a collection data structure and I asked it to add something, it should add whatever I asked it to add.

Thoughts @jeromelaban @dr1rrb ?

Reply:
In XAML, CalendarView use UTC +00:00:00.
But DateTime and DateTimeOffset are normally Local Time.
So I convert Input DateTime from local to UTC.

Reply:
For example, _April **28th** 10 p.m EST_ is same with _April **29th** 2 A.M UTC_. The number of days are different.
So user set the SelectedDates to 28th but he will see that 29th are selected  in the CalendarView.

Reply:
I know. But to solve this problem perfectly, I think we should change calendarview's time from UTC to local. As you can check, in Windows, the CalendarView uses local time zone, not UTC. But to change the time zone of CalendarView in Wasm will take a lot of cost in aspect of time and code. so I think we have to open other ticket about this.